src/webpack-angular-types-plugin/type-extraction/type-printing.ts

```diff
@@ -11,6 +11,23 @@ function truncateImportPart(typeStr: string): string {
     return typeStr.replace(regex, "");
 }
 
+/*
+ * Given an array of type strings, it is searched if both the "true" and "false"
+ * boolean literal types appear in the array. If so, they are replaced by "boolean"
+ */
+function replaceTrueFalseUnionByBooleanIfExists(union: string[]): string[] {
+    // replace true/false by boolean
+    const trueIndex = union.indexOf("true");
+    const falseIndex = union.indexOf("false");
+    if (trueIndex > -1 && falseIndex > -1) {
+        union = union.filter(
+            (elem, index) => index !== trueIndex && index !== falseIndex
+        );
+        union.push("boolean");
+    }
```

Comment:
I wonder if this can be simplified

You use `indexOf` twice before even start filtering the list again just to figure out if you need to push `boolean` 🤔 

Probably not performance relevant and I don't have a better idea right away

Reply:
I need to check if both indices exists before i can remove them. Only when i have both indices i can actually filter them out.
I could add a check right like `trueIndex > -1` before querying for the false index, which eventually saves me the second query if `trueIndex===-1` anyway.
Idk if this is worth to be optimized at the moment.

Reply:
Leave it as is